src/clj/game/cards/hardware.clj

```diff
@@ -796,6 +796,41 @@
                           :effect (effect (update! (dissoc card :Omnidrive-prog))
                                           (use-mu (:memoryunits target)))}}}
 
+   "Patchwork"
+   (letfn [(patchwork-discount [cost-type bonus-fn]
+             {:async true
+              :req (req (and (= "Runner" (:side target))
+                             ;; The card being installed/played is still in the hand; we therefore need at least 2
+                             ;; cards for this effect to make sense.
+                             (< 0 (count (remove (partial same-card? target) (:hand runner))))))
+              :effect (req (let [playing target]
+                             (continue-ability
+                               state side
+                               {:prompt (str "Trash a card to lower the " cost-type " cost of " (:title playing) " by 2 [Credits].")
+                                :choices {:req #(and (in-hand? %)
+                                                     (= "Runner" (:side %))
+                                                     (not (same-card? % playing)))}
+                                :msg (msg "trash " (:title target) " to lower the " cost-type " cost of "
+                                          (:title playing) " by 2 [Credits]")
+                                :effect (req (trash state side target {:unpreventable true})
+                                             (bonus-fn state side [:credit -2])
+                                             (unregister-events state side card))
+                                :cancel-effect (effect (effect-completed eid))}
+                               card nil)))})]
+     {:in-play [:memory 1]
+      :implementation "Click Patchwork before playing/installing a card."
+      :abilities [{:once :per-turn
+                   :effect (effect (register-events {:pre-play-instant (patchwork-discount "play" play-cost-bonus)
```

Comment:
So I think this would be better if the events were always "on", and we just set a flag in the card under `[:special :patchwork]` or something. The `:req` for the handler above would then require that flag to be set. I know a lot of cards to the register/unregister trick but it's pretty hacky.